src/backend.rs

```diff
@@ -238,11 +246,9 @@ impl io::Read for Mock {
 #[cfg(any(test, feature = "mock"))]
 impl io::Write for Mock {
 	fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
-		if let Some(err) = self.write_error.take() {
-			Err(err)
-		} else {
-			Ok(buf.len())
-		}
```

Comment:
This removes the ability to inject write errors into the mock, though the Mock API still supports it. I would like to retain this ability. Could you add your change within the existing `Ok` branch instead of remove the existing code?

Reply:
Thanks!